### PR TITLE
PAY-5360 added contact_details JSON in Refunds APIs request body

### PR DESCRIPTION
**Pull request by sayali-salunkhe**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsService.java [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-import java.math.BigDecimal;
+import uk.gov.hmcts.payment.api.dto.RetroSpectiveRemissionRequest;
 
 public interface PaymentRefundsService {
 
     ResponseEntity<RefundResponse> createRefund(PaymentRefundRequest paymentRefundRequest, MultiValueMap<String, String> headers);
 
-    ResponseEntity<RefundResponse> createAndValidateRetroSpectiveRemissionRequest(String remissionReference, MultiValueMap<String, String> headers);
+    ResponseEntity<RefundResponse> createAndValidateRetroSpectiveRemissionRequest(
+            RetroSpectiveRemissionRequest retroSpectiveRemissionRequest, MultiValueMap<String, String> headers);
```
Comment (YasharRahvar) [inline api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsService.java:15]: fix the spelling

Comment (sayali-salunkhe) [reply api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsService.java]: Done

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/OnlineCardPaymentFunctionalTest.java [hunk 2]
```diff
@@ -326,6 +322,35 @@ public void retrieveCMCCardPaymentTestShouldReturnAutoApportionedFees() {
         }));
     }
 
+    // TO BE IMPLEMENTED IN THE CARD PAYMENT SCOPE
```
Comment (pats-john) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/OnlineCardPaymentFunctionalTest.java:325]: Why is this test commented.

Comment (sinanghmcts) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/OnlineCardPaymentFunctionalTest.java:325]: This test was in RefundRequestorPBAFunctionalTets class and had an @ignore tag. I think this case fall into scope of OnlineCardPayment scope. I just moved it in this class to discuss further.

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/RefundsRequestorJourneyPBAFunctionalTest.java [hunk 3]
```diff
@@ -569,10 +570,12 @@ public void positive_add_remission_and_add_refund_for_2_pba_payments() {
...
+        RetrospectiveRemissionRequest retrospectiveRemissionRequest
+            = PaymentFixture.aRetroRemissionRequest(remissionReference);
+
         Response refundResponse = paymentTestService.postSubmitRefund(USER_TOKEN_PAYMENTS_REFUND_REQUESTOR_ROLE,
-            SERVICE_TOKEN_PAYMENT,
-            RetrospectiveRemissionRequest.retrospectiveRemissionRequestWith().remissionReference(remissionReference).contactDetails(
-                    ContactDetails.contactDetailsWith().build()).build());
+            SERVICE_TOKEN_PAYMENT, retrospectiveRemissionRequest);
+
         assertThat(refundResponse.statusCode()).isEqualTo(HttpStatus.CREATED.value());
```
Comment (pats-john) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/RefundsRequestorJourneyPBAFunctionalTest.java:578]: We need to add a negative test that checks on the Error scenarios of the Contact Details.make sure we are getting the right HTTPS Response Codes and reasons as well.

Comment (pats-john) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/RefundsRequestorJourneyPBAFunctionalTest.java:578]: A bare miniumum of the Contact Details as per Acceptance Criteria should be given...

**Review by pats-john — CHANGES_REQUESTED** [review]

**Pull request merged**